src/controllers/redeploy.go

```diff
@@ -8,9 +8,16 @@ func redeploy(callbackID string, data map[string]interface{}) {
 	stop(callbackID, data)
 	state := history.GetState(data["git_repo"].(string))
 	if state.Status == "stopped" {
-		data["subdomain"] = state.Subdomain
-		data["access"] = state.Access
-		data["server_name"] = state.Server
+		if _, ok := data["subdomain"]; !ok {
+			data["subdomain"] = state.Subdomain
+		}
+		if _, ok := data["access"]; !ok {
+			data["access"] = state.Access
+		}
+		if _, ok := data["server_name"]; !ok {
+			data["server_name"] = state.Server
+		}
```

Comment:
you also need to check if state.Subdomain, state.Access, state.Server are set or not. If not then raise an error.

Reply:
Cool,
Btw to display the original state parameters what @ankurshaswat was saying, can't we set them as value in the redeploy dialog ?

Reply:
no we cannot because when the dialog box is sent we don't know the service which will be redeployed.

Reply:
An empty string check should be sufficient here I guess ?

Reply:
yes it should. just && it with !ok